apis/kahumgr/v1beta1/schedulepolicy.go

```diff
@@ -0,0 +1,237 @@
+	}
+	var record1 []string
+	var record2 []string
+	strMap := make(map[time.Weekday]string)
+
+	for _, day := range w.Days {
+		// invalid format checking
+		if _, exist1 := DaysType[day]; !exist1 {
+			record1 = append(record1, day)
+		}
+		// duplicate checking
+		if _, exists := strMap[DaysType[day]]; exists {
+			record2 = append(record2, day)
+		} else {
+			strMap[DaysType[day]] = day
+		}
+	}
+
+	if len(record1) > 0 {
+		return fmt.Errorf("invalid day of the week: %v, in WeeklyPolicy you should provide the day"+
+			" as Sunday, Sun, Monday, Mon, TuesDay, Tue, Wednesday, Wed, Thursday, Thurs, Friday, Fri,"+
+			"Saturday, Sat", record1)
+	}
+	if len(record2) > 0 {
+		return fmt.Errorf("Duplictae day of the week: %v, in WeeklyPolicy ", record2)
+	}
+	return nil
+}
+
+// Monthly Policy contains the dates and time  in a month when the action should be triggered
+// the cron example  25 11 (1,5,8,11,18) * ?
+// (so on given dates every month at 11:25 Hrs triggers)
+type MonthlyPolicy struct {
+	// Dates of the month when action should be triggered. If given date does not exist in a month then rollover to the next
+	// date of month. Example 31 is specified then in Feb it will trigger on either 1st or 2nd March based on leap year or not.
+	// +kubebuilder:validation:Required
+	Dates []int `json:"dates"`
+	// eg 12:15
+	// +kubebuilder:validation:Required
+	Time string `json:"time"`
+	// +kubebuilder:validation:Maximum=256
+	// +kubebuilder:validation:Minimum=1
+	// +kubebuilder:default=12
+	// +kubebuilder:validation:Optional
+	MaxCopies int `json:"maxCopies"`
+}
+
+func (m *MonthlyPolicy) CheckTimeFormat() error {
+	return checkTimeFormat(m.Time)
+}
+
+// validates the MonthlyPolicy
+func (m *MonthlyPolicy) Validate() error {
+	err := m.CheckTimeFormat()
+	if err != nil {
+		return err
+	}
+	var record []int
+	for _, date := range m.Dates {
+		// invalid format checking
+		if date <= 0 || date > 31 {
+			record = append(record, date)
+		}
+	}
+
+	if len(record) > 0 {
+		return fmt.Errorf("invalid date of the Month: %v, in MonthlyPolicy you should provide the date"+
+			" as (0,31] only", record)
+	}
+	intMap := make(map[int]int)
+	// validate for duplicate values
+	for _, v := range m.Dates {
+		// duplicate checking
+		if _, exists := intMap[v]; exists {
+			record = append(record, v)
+		} else {
+			intMap[v] = v
+		}
+	}
+
+	if len(record) > 0 {
+		return fmt.Errorf("duplicate date of the Month: %v, in MonthlyPolicy you should provide the date"+
+			" as (0,31] only", record)
+	}
+	return nil
+}
+
+// SchedulePolicyspec
+type SchedulePolicySpec struct {
+	Hourly  *HourlyPolicy  `json:"hourly,omitempty"`
+	Daily   *DailyPolicy   `json:"daily,omitempty"`
+	Weekly  *WeeklyPolicy  `json:"weekly,omitempty"`
+	Monthly *MonthlyPolicy `json:"monthly,omitempty"`
+}
+
+// SchedulePolicy is the Schema for the policy API
+// +genclient
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:scope=Namespaced
+// +kubebuilder:subresource:status
```

Comment:
IMP, ```SchedulePolicy``` does not maintain ```status```

Reply:
it will not have any impact, but any how changed